dicer/mappers/observation_mapper.py

```diff
@@ -0,0 +1,126 @@
+                                 indexed_dimensions: List[DimensionDeclaration],
+                                 modifier_dimensions: List[DimensionDeclaration]) -> Optional[TLObservationMetadata]:
+        metadata = None
+        for modifier_dim in modifier_dimensions:
+            metadata_values = dict()
+            modifier_dimension_id = self.get_id_by_dimension_name(indexed_dimensions, modifier_dim.name)
+            modifier_dimension_element_id = cell.dimensionIndexes[modifier_dimension_id]
+            if modifier_dimension_element_id is not None:
+                modifier = next(filter(lambda x: x.modifier_code == modifier_dim.modifierCode, self.modifiers))
+                modifier_value = hypercube.dimensionElements[modifier_dim.name][modifier_dimension_element_id]
+                metadata_values[modifier] = value_by_value_type(modifier_value, modifier.value_type)
+                metadata = TLObservationMetadata(metadata_values)
+        return metadata
+
+    def get_observation_trial_visit(self,
+                                    cell: Cell,
+                                    hypercube: Hypercube,
+                                    trial_visits_dimension_id: int) -> Optional[TLTrialVisit]:
+        trial_visit = None
+        trial_visit_dim_element_id = cell.dimensionIndexes[trial_visits_dimension_id]
+        if trial_visit_dim_element_id is not None:
+            trial_visit_dim_element = TrialVisitDimensionElement(
+                **hypercube.dimensionElements['trial visit'][trial_visit_dim_element_id])
+            trial_visit = next(filter(lambda x:
+                                      (x.study.study_id, x.rel_time_label) == (trial_visit_dim_element.studyId,
+                                                                               trial_visit_dim_element.relTimeLabel),
+                                      self.trial_visits))
+        return trial_visit
+
+    def get_observation_visit(self, cell: Cell, hypercube: Hypercube, visit_dimension_id: int) -> Optional[TLVisit]:
+        visit = None
+        visit_dim_element_id = cell.dimensionIndexes[visit_dimension_id]
+        if visit_dim_element_id is not None:
+            visit_dim_element = VisitDimensionElement(**hypercube.dimensionElements['visit'][visit_dim_element_id])
+            visit = next(filter(lambda x: x.identifier == visit_dim_element.encounterIds['VISIT_ID'], self.visits))
+        return visit
+
+    def get_observation_concept(self, cell: Cell, hypercube: Hypercube, concept_dimension_id: int) -> TLConcept:
+        concept_dim_element = ConceptDimensionElement(
+            **hypercube.dimensionElements['concept'][cell.dimensionIndexes[concept_dimension_id]])
+        concept = next(filter(lambda x: x.concept_code == concept_dim_element.conceptCode, self.concepts))
+        return concept
+
+    def get_observation_patient(self, cell: Cell, hypercube: Hypercube, patient_dimension_id: int) -> TLPatient:
```

Comment:
Instead of passing the hypercube, you could also pass the list of PatientDimensionElements as an argument.